main.rb

```diff
@@ -0,0 +1,83 @@
+module Enumerable
+  def my_each(&prc)
+    i = 0
+    while i < length
+      if is_a?(Array)
+        yield self[i]
+      elsif prc.arity == 1
+        yield(assoc keys[i])
+      else
+        yield(keys[i], self[keys[i]])
+      end
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    idx = 0
+    while idx < length
+      yield(self[idx], idx)
+      idx += 1
+    end
+    self
+  end
+
+  def my_select
+    if is_a?(Array)
+      selected = []
+      my_each { |ele| selected << ele if yield ele }
+    else
+      selected = {}
+      my_each { |key, value| selected[key] = value if yield(key, value) }
+    end
+    selected
+  end
+
+  def my_all?(&prc)
+    my_each do |ele|
+      return false if prc.call(ele) == false
+    end
+    true
+  end
+
+  def my_any?(&prc)
+    my_each do |ele|
+      return true if prc.call(ele) == true
+    end
+    false
+  end
```

Comment:
`my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- when no block or argument is given returns true if at least one of the collection is not false or nil
```ruby
[1, 2, 3].my_any? #should return true
[1, 2, nil].my_any? #should return true
[false, false, nil].my_any? #should return false
```
- when a class is passed as an argument returns true if at least one of the collection is a member of such class
```ruby
[1, 2, 3].my_any?(Integer) #should return true
[1, 'demo', false].my_ant?(Integer) #should return true
['demo', false, nil].my_any?(Integer) #should return false
```
- when a Regex is passed as an argument returns false if none of the collection matches the Regex
```ruby
%w[dog door rod blade].my_any?(/d/)   #should return true.
```
- when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
```ruby
[1, 2, 3].my_any?(3)   #should return true
```